src/test/url-loader/url-resolver_test.ts

```diff
@@ -39,6 +39,20 @@ class SimplestUrlResolver extends UrlResolver {
 }
 
 suite('UrlResolver', () => {
+  suite('resolve', () => {
+    const resolver = new SimplestUrlResolver();
+    function resolve(url: string, baseUrl: string) {
+      return resolver.resolve(url as FileRelativeUrl, baseUrl as ResolvedUrl);
+    }
+    test('can resolve a url when relative url contains no pathname', () => {
+      assert.equal(resolve('', '/foo.html?fiz#buz'), '/foo.html?fiz');
+      assert.equal(resolve('#fiz', '/foo.html'), '/foo.html#fiz');
+      assert.equal(resolve('#fiz', '/foo.html#buz'), '/foo.html#fiz');
+      assert.equal(resolve('?fiz', '/foo.html'), '/foo.html?fiz');
+      assert.equal(resolve('?fiz', '/foo.html?buz'), '/foo.html?fiz');
```

Comment:
Could use a test that just a query string overrides the fragment in the base

```typescript
      assert.equal(
          resolve(
              `?fiz`, `/foo.html?bar#buz`), `/foo.html?fiz`);
```

Reply:
Added.